### ISSUE TITLE
Store SP in Database

### ISSUE THREAD

**Issue by juliardi** [codegreencreative/laravel-samlidp#74]
Is it possible to store the list of Service Provider in the database and grab it from there whenever someone login/logout ?
Or is it a security issue?

**Comment by upwebdesign** [comment]
@juliardi I suppose you could store them in a database. There is a lot of sensitive information being stored in databases. I don't see why this would be any different. Whether or not your database is secure is up to you. You should be able to set the config to your liking using the config helper in Laravel.

**Comment by juliardi** [comment]
@upwebdesign Is this library currently support fetching SP list from database ? if it is, how do I config it?

**Comment by upwebdesign** [comment]
@juliardi please see the [config section of the readme](https://github.com/codegreencreative/laravel-samlidp#config).

**Comment by juliardi** [comment]
The config section only stated that the SP list stored in the config file, so I conclude currently this library doesn't support fetching SP list from database, but storing the SP list itself in the database is something allowable but would need a modification to this package. Thank you for your answer @upwebdesign.

**Comment by upwebdesign** [comment]
@juliardi yes, you are correct. This package does not support DB storage for SP's. This could possibly be a feature request.

**Comment by jrgilman** [comment]
@upwebdesign we are actually in the process of doing this for a piece of software we're working on and are curious if you're open to merging a PR from us for the feature.

We're debating about whether or not we want to do this by providing a ServiceProvider eloquent model in the package itself - or providing an abstract class and allowing the user to provide their own implementation as a config variable. If you have a preference let me know as that will effect which way we want to go with this - we would like to get this PR'ed into the package.

We would also make sure to provide a test suite that ensures it doesn't break any existing features that it touches.

**Comment by upwebdesign** [comment]
@jrgilman, involving the database with this package would expose a potential breaking change. If a database is considered, all SP information should be able to be stored in the database while still allowing for .env/config implementation. If your team would like to tackle a DB approach, that would be cool.
